From ee1dc8d8b9fa43c76d3a5abb4e8cd9c8aaba11d1 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Marc-Andr=C3=A9=20Lureau?= <marc-andre.lureau@nokia.com>
Date: Wed, 29 Jul 2009 15:33:19 +0300
Subject: [PATCH 85/85] sink-input: don't make volume relative to dest sink reference volume

When sink-input is moving to a new sink, PulseAudio keep the relative
volume and multiply with the reference volume of the new sink.  But
that will cause volume to change to 100% if the dest sink reference
volume is 100% and the sink-input was single on the original sink.

User will experience volume at 100% after that, because stream-restore
will remember that this kind of sink-input was 100%.

For now, let's disable relative * reference volume change when moving
sink-input.
---
 src/pulsecore/sink-input.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/pulsecore/sink-input.c b/src/pulsecore/sink-input.c
index 06d9b74..6d450c2 100644
--- a/src/pulsecore/sink-input.c
+++ b/src/pulsecore/sink-input.c
@@ -1150,11 +1150,13 @@ int pa_sink_input_start_move(pa_sink_input *i) {
     if (i->sink->flags & PA_SINK_FLAT_VOLUME) {
         pa_cvolume new_volume;
 
+#ifndef NOKIA
         /* Make the virtual volume relative */
         pa_sink_input_get_relative_volume(i, &i->virtual_volume);
 
         /* And reset the the relative volume */
         pa_sink_input_set_relative_volume(i, NULL);
+#endif
 
         /* We might need to update the sink's volume if we are in flat
          * volume mode. */
@@ -1245,10 +1247,12 @@ int pa_sink_input_finish_move(pa_sink_input *i, pa_sink *dest, pa_bool_t save) {
     if (i->sink->flags & PA_SINK_FLAT_VOLUME) {
         pa_cvolume new_volume;
 
+#ifndef NOKIA
         /* Make relative volume absolute again */
         pa_cvolume t = dest->reference_volume;
         pa_cvolume_remap(&t, &dest->channel_map, &i->channel_map);
         pa_sw_cvolume_multiply(&i->virtual_volume, &i->virtual_volume, &t);
+#endif
 
         /* We might need to update the sink's volume if we are in flat volume mode. */
         pa_sink_update_flat_volume(i->sink, &new_volume);
-- 
1.6.3.3

